hclogslog/adapter.go

```diff
@@ -0,0 +1,204 @@
+		if key == "" {
+			if a.Value.Equal(slog.Value{}) {
+				return nil
+			}
+
+			// If the key is empty but there is a value, then make the key
+			// the position of the attributes in the record.
+			key = strconv.Itoa(pos)
+		}
+
+		attrs = append(attrs, prefix+key, val.Any())
+	}
+
+	return attrs
+}
+
+// Handle handles the Record.
+// It will only be called when Enabled returns true.
+// The Context argument is as for Enabled.
+// It is present solely to provide Handlers access to the context's values.
+// Canceling the context should not affect record processing.
+// (Among other things, log messages may be necessary to debug a
+// cancellation-related problem.)
+//
+// Handle methods that produce output should observe the following rules:
+//   - If r.Time is the zero time, ignore the time.
+//   - If r.PC is zero, ignore it.
+//   - Attr's values should be resolved.
+//   - If an Attr's key and value are both the zero value, ignore the Attr.
+//     This can be tested with attr.Equal(Attr{}).
+//   - If a group's key is empty, inline the group's Attrs.
+//   - If a group has no Attrs (even if it has a non-empty key),
+//     ignore it.
+func (h *Handler) Handle(_ context.Context, rec slog.Record) error {
+	attrs := make([]any, 0, rec.NumAttrs()*2)
+
+	var cnt int
+
+	rec.Attrs(func(a slog.Attr) bool {
+		attrs = append(attrs, h.processAttr(cnt, h.prefix, a)...)
+		cnt++
+		return true
+	})
+
+	h.l.Log(h.translateLevel(rec.Level), rec.Message, attrs...)
+	return nil
+}
+
+// WithAttrs returns a new Handler whose attributes consist of
+// both the receiver's attributes and the arguments.
+// The Handler owns the slice: it may retain, modify or discard it.
+
+func (h *Handler) WithAttrs(slogAttrs []slog.Attr) slog.Handler {
+	attrs := make([]any, 0, len(slogAttrs))
+
+	for i, a := range slogAttrs {
+		attrs = append(attrs, h.processAttr(i, h.prefix, a)...)
+	}
+
+	return &Handler{
+		l:      h.l.With(attrs...),
+		prefix: h.prefix,
+	}
+}
+
+// WithGroup returns a new Handler with the given group appended to
+// the receiver's existing groups.
+// The keys of all subsequent attributes, whether added by With or in a
+// Record, should be qualified by the sequence of group names.
+//
+// How this qualification happens is up to the Handler, so long as
+// this Handler's attribute keys differ from those of another Handler
+// with a different sequence of group names.
+//
+// A Handler should treat WithGroup as starting a Group of Attrs that ends
+// at the end of the log event. That is,
+//
+//	logger.WithGroup("s").LogAttrs(level, msg, slog.Int("a", 1), slog.Int("b", 2))
+//
+// should behave like
+//
+//	logger.LogAttrs(level, msg, slog.Group("s", slog.Int("a", 1), slog.Int("b", 2)))
+//
+// If the name is empty, WithGroup returns the receiver.
+func (h *Handler) WithGroup(name string) slog.Handler {
+	prefix := h.prefix
+
+	if name != "" {
+		prefix = h.prefix + name + "."
+	}
+
+	return &Handler{
+		l:      h.l,
+		prefix: prefix,
+	}
```

Comment:
We do not exactly return the receiver here, it's a pointer to a new handler with the same settings.

Maybe this should be:
```
if name == "" {
    return h
}
// ...
```